apps/koala-frontend/src/app/shared/dates.validator.ts

```diff
@@ -0,0 +1,14 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+export const datesStartEndValidator: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
+  const startDate: Date = control.get('start')?.value;
+  const endDate: Date = control.get('end')?.value;
+
+  if (startDate && endDate) {
+    return startDate <= endDate ? null : { datesWrong: true };
+  } else if (startDate || endDate) {
+    return { datesNotCompleted: true };
```

Comment:
Not sure why you distinguish between wrong and incomplete if you only check for datesWrong in the ui.

Reply:
In case of incomplete there is no error right now (you just cannot save because the button is disabled). But you also cannot check if start is smaller than end because if you just fill end the condition is still true. That's why this is considered invalid (button is disabled), but is distinguished from the error scenario to not show an error. Does that make sense?